src/order/preorder_hom.lean

```diff
@@ -29,30 +29,28 @@ instance : has_coe_to_fun (preorder_hom α β) :=
 { F := λ f, α → β,
   coe := preorder_hom.to_fun }
 
+initialize_simps_projections preorder_hom (to_fun → coe)
+
 @[mono]
 lemma monotone (f : α →ₘ β) : monotone f :=
 preorder_hom.monotone' f
 
-@[simp]
-lemma coe_fun_mk {f : α → β} (hf : _root_.monotone f) (x : α) : mk f hf x = f x := rfl
-
-@[ext] lemma ext (f g : preorder_hom α β) (h : ∀ a, f a = g a) : f = g :=
-by { cases f, cases g, congr, funext, exact h _ }
+@[simp] lemma to_fun_eq_coe {f : α →ₘ β} : f.to_fun = f := rfl
+@[simp] lemma coe_fun_mk {f : α → β} (hf : _root_.monotone f) : (mk f hf : α → β) = f := rfl
 
-lemma coe_inj (f g : preorder_hom α β) (h : (f : α → β) = g) : f = g :=
-by { ext, rw h }
+@[ext] -- See library note [partially-applied ext lemmas]
+lemma ext (f g : preorder_hom α β) (h : (f : α → β) = g) : f = g :=
+by { cases f, cases g, congr, exact h }
 
 /-- The identity function as bundled monotone function. -/
-@[simps]
+@[simps {fully_applied := ff}]
 def id : preorder_hom α α :=
 ⟨id, monotone_id⟩
 
 instance : inhabited (preorder_hom α α) := ⟨id⟩
 
-@[simp] lemma coe_id : (@id α _ : α → α) = id := rfl
-
 /-- The composition of two bundled monotone functions. -/
-@[simps]
+@[simps {fully_applied := ff}]
```

Comment:
Isn't the error just that the `simps` lemma for this definition has been renamed from `comp_to_fun` to `comp_coe`?

Reply:
That's the error that CI shows (and as you say it's easily fixed) but there are `tidy` incantations further below which I couldn't fix - I'll try to amend so CI shows it more clearly!

Reply:
Alright, changed - the issue is that `skeletal_functor` now times out when it didn't previously